network/phonebook/phonebook.go

```diff
@@ -29,23 +29,54 @@ import (
 // of how many addresses the phonebook actually has. ( with the retry-after logic applied )
 const getAllAddresses = math.MaxInt32
 
-// PhoneBookEntryRoles defines the roles that a single entry on the phonebook can take.
+// Roles defines the roles that a single entry on the phonebook can take.
 // currently, we have two roles : relay role and archival role, which are mutually exclusive.
 //
-//msgp:ignore PhoneBookEntryRoles
-type PhoneBookEntryRoles int
+//msgp:ignore Roles
+type Roles struct {
+	roles role
+	_     func() // func is not comparable so that Roles. This is to prevent roles misuse and direct comparison.
+}
+
+var (
+	// PhoneBookEntryRelayRole used for all the relays that are provided either via the algobootstrap SRV record
+	// or via a configuration file.
+	PhoneBookEntryRelayRole = Roles{roles: relayRole}
```

Comment:
I like the change, above, from `PhoneBookEntryRoles` to `Roles`, so that users of this code can say `phonebook.Roles`.  I'd like to see `PhoneBookEntryRole` changed to `Role` for the same reason. I think if it were new code, the linter would complain about its current name.